clic/src/statistics.cpp

```diff
@@ -0,0 +1,324 @@
+  tier1::crop_func(device, sum_per_label, result_vector, offset, 7, 0, num_measurements, 1, 1);
+  result_vector->read(sum_intensity.data());
+  region_props["sum_intensity"] = sum_intensity;
+
+  std::vector<float> area(num_measurements);
+  tier1::crop_func(device, sum_per_label, sum_dim, offset, 3, 0, num_measurements, 1, 1);
+  sum_dim->read(area.data());
+  region_props["area"] = area;
+
+  std::vector<float> mean_intensity(num_measurements);
+  tier1::paste_func(device, sum_dim, label_statistics_image, offset, 7, 0);
+  tier1::divide_images_func(device, result_vector, sum_dim, avg_dim);
+  avg_dim->read(mean_intensity.data());
+  region_props["mean_intensity"] = mean_intensity;
+
+  tier1::paste_func(device, avg_dim, label_statistics_image, offset, 6, 0);
+
+  // Sum intensity times x, y, z and mass center
+
+  std::vector<std::string> dim_names = { "x", "y", "z" };
+  tier1::crop_func(device, sum_per_label, result_vector, offset, 4 + 3, 0, num_measurements, 1, 1);
+  for (int dim = 0; dim < 3; ++dim)
+  {
+    tier1::crop_func(device, sum_per_label, sum_dim, offset, 4 + dim, 0, num_measurements, 1, 1);
+
+    std::vector<float> sum_intensity_times(num_measurements);
+    sum_dim->read(sum_intensity_times.data());
+    region_props["sum_intensity_times_" + dim_names[dim]] = sum_intensity_times;
+
+    tier1::divide_images_func(device, sum_dim, result_vector, avg_dim);
+
+    std::vector<float> mass_center(num_measurements);
+    avg_dim->read(mass_center.data());
+    region_props["mass_center_" + dim_names[dim]] = mass_center;
+    tier1::paste_func(device, avg_dim, label_statistics_image, offset, 3 + dim, 0);
+  }
+
+  // Sum x, y, z and centroid
+  tier1::crop_func(device, sum_per_label, result_vector, offset, 3, 0, num_measurements, 1, 1);
+  for (int dim = 0; dim < 3; ++dim)
+  {
+    tier1::crop_func(device, sum_per_label, sum_dim, offset, dim, 0, num_measurements, 1, 1);
+
+    std::vector<float> sum(num_measurements);
+    sum_dim->read(sum.data());
+    region_props["sum_" + dim_names[dim]] = sum;
+    tier1::divide_images_func(device, sum_dim, result_vector, avg_dim);
+
+    std::vector<float> centroid(num_measurements);
+    avg_dim->read(centroid.data());
+    region_props["centroid_" + dim_names[dim]] = centroid;
+    tier1::paste_func(device, avg_dim, label_statistics_image, offset, 3 + dim, 0);
+  }
+
+  // Second part: determine parameters which depend on other parameters
+  std::cout << "nb_labels: " << nb_labels << std::endl;
+  auto label_statistics_stack = Array::create(nb_labels, height, 6, 3, dType::FLOAT, mType::BUFFER, device);
+  label_statistics_stack->fill(0);
+
+  cle::print<float>(label_statistics_image, "label_statistics_image");
+
+  const KernelInfo    kernel_std = { "standard_deviation_per_label", kernel::standard_deviation_per_label };
+  const RangeArray    range_std = { 1, static_cast<size_t>(height), 1 };
+  const ParameterList params_std = { { "src_statistics", label_statistics_image },
```

Comment:
```suggestion
  ParameterList params_std = { { "src_statistics", label_statistics_image },
```

Reply:
cannot be const because we need to update the `z` parameter in the incoming loop